lib/routes/syosetu/ranking-r18.ts

```diff
@@ -0,0 +1,258 @@
+                target: `/rankingr18/${domain}/${period}_${NovelType.TOTAL}/limit=5`,
+            }))
+        );
+
+export const route: Route = {
+    path: '/rankingr18/:sub/:type/:routeParams?',
+    categories: ['reading'],
+    example: '/syosetu/rankingr18/noc/daily_total/limit=50',
+    parameters: getParameters(),
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: 'R18 Rankings',
+    maintainers: ['SnowAgar25'],
+    handler,
+    description: `
+| Period | Description | 説明 |
+| --- | --- | --- |
+| daily | Daily Ranking | 日間ランキング |
+| weekly | Weekly Ranking | 週間ランキング |
+| monthly | Monthly Ranking | 月間ランキング |
+| quarter | Quarterly Ranking | 四半期ランキング |
+| yearly | Yearly Ranking | 年間ランキング |
+
+| Novel Type | Description | 説明 |
+| --- | --- | --- | --- |
+| total | All Works | 総合 |
+| t | Short Stories | 短編 |
+| r | Ongoing Series | 連載中 |
+| er | Completed Series | 完結済 |
+
+Note: Combine Period and Novel Type with \`_\`. For example: \`daily_total\`, \`weekly_r\`, \`monthly_er\``,
+    radar: [
+        {
+            source: ['noc.syosetu.com/rank/list/type/:type'],
+            target: '/rankingr18/noc/:type',
+        },
+        {
+            source: ['mid.syosetu.com/rank/list/type/:type'],
+            target: '/rankingr18/mid/:type',
+        },
+        {
+            source: ['mnlt.syosetu.com/rank/list/type/:type'],
+            target: '/rankingr18/mnlt/:type',
+        },
+        {
+            source: ['mnlt.syosetu.com/rank/bllist/type/:type'],
+            target: '/rankingr18/mnlt-bl/:type',
+        },
+        ...getBest5RadarItems(),
+    ],
+};
+
+function parseRankingType(type: string): { period: RankingPeriod; novelType: NovelType } {
+    const [periodStr, novelTypeStr] = type.split('_');
+
+    const period = periodStr as RankingPeriod;
+    const novelType = novelTypeStr as NovelType;
+
+    if (!Object.values(RankingPeriod).includes(period) || !Object.values(NovelType).includes(novelType)) {
+        throw new InvalidParameterError(`Invalid ranking type: ${type}`);
+    }
+
+    return {
+        period: periodStr as RankingPeriod,
+        novelType: novelTypeStr as NovelType,
+    };
+}
+
+function getRankingTitle(type: string): string {
+    const { period, novelType } = parseRankingType(type);
+    return `${periodToJapanese[period]}${novelTypeToJapanese[novelType]}ランキング`;
+}
+
+async function handler(ctx: Context): Promise<Data> {
+    const { sub, type } = ctx.req.param();
+    const baseUrl = `https://${sub === 'mnlt-bl' ? 'mnlt' : sub}.syosetu.com`;
+    const rankingUrl = `${baseUrl}/rank/list/type/${type}`;
+    const api = new NarouNovelFetch();
+
+    const routeParams = querystring.parse(ctx.req.param('routeParams'));
+    const limit = Number(routeParams.limit || 300);
```

Comment:
Please make use of the [common parameter limit](https://docs.rsshub.app/guide/parameters#limit-entries) `ctx.req.query('limit')` instead.